sonar-kotlin-plugin/src/main/java/org/sonarsource/kotlin/checks/EmptyLineRegexCheck.kt

```diff
@@ -0,0 +1,258 @@
+                    getSecondariesForToRegex(callExpression, kotlinFileContext.bindingContext)
+                }
+                else -> emptyList<KtElement>() to null
+            }
+            if (main != null && secondaries.isNotEmpty()) {
+                kotlinFileContext.reportIssue(
+                    main,
+                    MESSAGE,
+                    kotlinFileContext.locationListOf(*secondaries.map { it to "This string can be empty." }.toTypedArray()),
+                )
+            }
+        }
+    }
+
+    private fun getSecondariesForPattern(callExpression: KtCallExpression, bindingContext: BindingContext): List<KtElement> {
+        return when (val preParent = callExpression.parent?.parent) {
+            is KtProperty -> {
+                preParent.usages()
+                    .mapNotNull { getStringInMatcherFind(it, bindingContext) }
+                    .filter { it.canBeEmpty(bindingContext) }
+            }
+            is KtDotQualifiedExpression -> {
+                getStringInMatcherFind(preParent.selectorExpression!!, bindingContext)
+                    ?.let { if (it.canBeEmpty(bindingContext)) listOf(it) else emptyList()} ?: emptyList()
+            }
+            else -> {
+                emptyList()
+            }
+        }
+    }
+
+    private fun getSecondariesForRegex(callExpression: KtCallExpression, bindingContext: BindingContext): List<KtElement> {
+        return when (val parent = callExpression.parent) {
+            is KtProperty -> {
+                parent.usages()
+                    .mapNotNull { getStringInRegexFind(it, bindingContext) }
+                    .filter { it.canBeEmpty(bindingContext) }
+            }
+            is KtDotQualifiedExpression -> {
+                getStringInRegexFind(parent.selectorExpression!!, bindingContext)
+                    ?.let { if (it.canBeEmpty(bindingContext)) listOf(it) else emptyList()} ?: emptyList()
+            }
+            else -> {
+                emptyList()
+            }
+        }
+    }
+
+    private fun getSecondariesForToRegex(callExpression: KtCallExpression, bindingContext: BindingContext): Pair<List<KtElement>, KtElement?> {
+        return (when (val parent = callExpression.parent?.parent) {
+            is KtProperty -> {
+                parent.usages()
+                    .mapNotNull { getStringInRegexFind(it, bindingContext) }
+                    .filter { it.canBeEmpty(bindingContext) }
+            }
+            is KtDotQualifiedExpression -> {
+                getStringInRegexFind(parent.selectorExpression!!, bindingContext)
+                    ?.let { if (it.canBeEmpty(bindingContext)) listOf(it) else emptyList()} ?: emptyList()
+            }
+            else -> {
+                emptyList()
+            }
+        }) to (callExpression.parent as? KtExpression)?.getResolvedCall(bindingContext)?.getReceiverExpression()
+    }
```

Comment:
The `when` blocks in these three methods seem to be pretty much the same. I think we can simplify this significantly.